### ISSUE TITLE
False `import-private-name` positive for relative imports

### ISSUE THREAD

**Issue by scop** [PyCQA/pylint#7078]
### Bug description

# pylint: disable=disallowed-name,relative-beyond-top-level,unused-import,missing-module-docstring
from .models import _Private
from ..models import _OtherPrivate

### Command used

```shell
pylint --enable-all-extensions foo.py
```

### Pylint output

```shell
************* Module foo
foo.py:2:0: C2701: Imported private object (_Private) (import-private-name)
foo.py:3:0: C2701: Imported private object (_OtherPrivate) (import-private-name)
```

### Expected behavior

No `import-private-name` messages.

I think relative imports should be exempt from this check because they do not tend to refer to external things which are documented and intended to not get this report.

### Pylint version

```shell
pylint 2.14.3
astroid 2.11.6
Python 3.8.12 (default, Sep  4 2021, 14:14:03) 
[GCC 9.3.0]
```
